scheduler/scheduler.go

```diff
@@ -28,21 +30,26 @@ type scheduledSession struct {
 }
 
 type scheduledInstance struct {
-	instance *types.Instance
-	ticker   *time.Ticker
-	cancel   context.CancelFunc
-	fails    int
+	instance     *types.Instance
+	playgroundId string
+	ticker       *time.Ticker
+	cancel       context.CancelFunc
+	fails        int
 }
 
 type scheduler struct {
 	scheduledSessions  map[string]*scheduledSession
 	scheduledInstances map[string]*scheduledInstance
 	tasks              map[string]Task
+	playgrounds        map[string]*types.Playground
+	playgroundTasks    map[string][]Task
 	started            bool
+	ticker             *time.Ticker
 
 	storage storage.StorageApi
 	event   event.EventApi
 	pwd     pwd.PWDApi
+	mx      sync.Mutex
```

Comment:
For what I see in the a PR, a single scheduler can handle multiple plagrounds, so shouldn't this mutex be per-playground?. The way it's implemented, every time it's used it'll hold on all the playgrounds operations.

Reply:
How are you going to do it? This is to update the managed playgrounds and matched tasks. Send away your suggestions

Reply:
Add a mutex per playground instead one mutex for all?

Reply:
And how do you update the playground list in a way that is safe?